### PR TITLE
Implemented dns_opt

### PR DESCRIPTION
**Pull request by Knetic**
Implements #3184. Adds the `dns_opt` feature to docker-compose, which lets users specify DNS options in their manifest which will get appended to the `/etc/resolv.conf` of the resultant container(s).

Relies on https://github.com/docker/docker-py/pull/1131, which was recently merged (and works locally, though I'm not sure how dependencies will work in the main docker-compose repo).

### CODE REVIEW

**Comment by Knetic** [comment]
This is only failing because the docker-py dependency isn't up to date with latest. I'm not sure if revving that dependency version in this PR is preferred, or if i should rebase after a maintainer has upped the version in this repo.

**Comment by aanand** [comment]
Up to you!

Whatever happens, this PR can't be merged until https://github.com/docker/docker-py/pull/1131 has made it into a release and `requirements.txt` and `setup.py` have been updated in master. But if you're happy to temporarily set the docker-py requirement to a Git URL, it'll help get to green CI earlier and ensure that there aren't any other hidden test failures.

**Comment by Knetic** [comment]
I've used it locally with the master branch of docker-py, and verified (both with the included tests and actual containers that I manually spawn) that it works, so I'm comfortable letting this PR sit until this repo's docker-py dependency is revved. Once the dependency is updated i can rebase and we can go from there.

Is there a rough timeline for when the 1.10 of docker-py will make it into this repo? Not sure if there's like, a "release train" or something, where downstreams get revved on a schedule after upstreams (like docker-py) are released.

File: docs/compose-file.md [hunk 1]
```diff
@@ -744,7 +744,7 @@ then read-write will be used.
 >     - container_name
 >     - container_name:rw
 
-### cpu\_shares, cpu\_quota, cpuset, domainname, hostname, ipc, mac\_address, mem\_limit, memswap\_limit, mem\_swappiness, oom\_score\_adj, privileged, read\_only, restart, shm\_size, stdin\_open, tty, user, working\_dir
+### cpu\_shares, cpu\_quota, cpuset, domainname, dns\_opts, hostname, ipc, mac\_address, mem\_limit, mem\_swappiness, memswap\_limit, oom\_score\_adj, privileged, read\_only, restart, shm\_size, stdin\_open, tty, user, working\_dir
```
Comment (joshwget) [inline docs/compose-file.md]: Shouldn't this be `dns_opt` here and not `dns_opts` to be consistent with the rest of the PR?

**Comment by shin-** [comment]
Rebased and merged in #4572

**Pull request closed without merging**